From 3769c83a06008a7079ff754ebfb5899c9dfdc963 Mon Sep 17 00:00:00 2001
From: HuyLe <hule@amperecomputing.com>
Date: Mon, 18 Jul 2022 14:45:31 +0700
Subject: [PATCH 1/3] button-handler: Add button event log

Support the Redfish log once the power button and the reset
button were pressed.

Signed-off-by: HuyLe <hule@amperecomputing.com>
---
 src/button_handler.cpp | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/src/button_handler.cpp b/src/button_handler.cpp
index 054f642..6ed5c55 100644
--- a/src/button_handler.cpp
+++ b/src/button_handler.cpp
@@ -5,6 +5,7 @@
 #include <phosphor-logging/lg2.hpp>
 #include <xyz/openbmc_project/State/Chassis/server.hpp>
 #include <xyz/openbmc_project/State/Host/server.hpp>
+#include <systemd/sd-journal.h>
 namespace phosphor
 {
 namespace button
@@ -32,6 +33,9 @@ constexpr auto mapperObjPath = "/xyz/openbmc_project/object_mapper";
 constexpr auto mapperService = "xyz.openbmc_project.ObjectMapper";
 constexpr auto BMC_POSITION = 0;
 
+std::string message;
+std::string redfishMsgId;
+
 Handler::Handler(sdbusplus::bus_t& bus) : bus(bus)
 {
     try
@@ -295,6 +299,11 @@ void Handler::powerReleased(sdbusplus::message_t& /* msg */)
     try
     {
         handlePowerEvent(PowerEvent::powerReleased);
+
+        message = "The power button pressed";
+        redfishMsgId = "OpenBMC.0.1.PowerButtonPressed";
+        sd_journal_send("MESSAGE=%s", message.c_str(),
+                "REDFISH_MESSAGE_ID=%s", redfishMsgId.c_str(), NULL);
     }
     catch (const sdbusplus::exception_t& e)
     {
@@ -307,6 +316,11 @@ void Handler::longPowerPressed(sdbusplus::message_t& /* msg */)
     try
     {
         handlePowerEvent(PowerEvent::longPowerPressed);
+
+        message = "The power button long pressed";
+        redfishMsgId = "OpenBMC.0.1.PowerButtonPressed";
+        sd_journal_send("MESSAGE=%s", message.c_str(),
+                "REDFISH_MESSAGE_ID=%s", redfishMsgId.c_str(), NULL);
     }
     catch (const sdbusplus::exception_t& e)
     {
@@ -320,6 +334,11 @@ void Handler::resetReleased(sdbusplus::message_t& /* msg */)
     try
     {
         handlePowerEvent(PowerEvent::resetReleased);
+
+        message = "The reset button pressed";
+        redfishMsgId = "OpenBMC.0.1.ResetButtonPressed";
+        sd_journal_send("MESSAGE=%s", message.c_str(),
+                "REDFISH_MESSAGE_ID=%s", redfishMsgId.c_str(), NULL);
     }
     catch (const sdbusplus::exception_t& e)
     {
-- 
2.25.1

